### DISCUSSION TITLE
SAR and Debian

### DISCUSSION THREAD

**Discussion by hack3rcon** [sysstat/sysstat#387] [Q&A]
Hello,
Does SAR automatically collect system activities or do you need to write a cron file?

Thank you.

**Comment by sysstat** [comment]
Hi,
You need a cron file or a systemd unit to trigger the data collector.
Note that you can ask sysstat to create and install it for you automatically if you configure, compile and install sysstat from the sources.

**Reply by hack3rcon** [reply]
Hi,
I installed it via **apt** command and then:
```
# systemctl enable sysstat
#
# systemctl start sysstat
```

Are these enough?

**Comment by sysstat** [comment]
The `sysstat` service should be started whenever the system is restarted to insert a LINUX RESTART message in the `saDD` datafile.
But you should also start the `sysstat-collect` service to collect data.
Don't know exactly what you should do in your case as I don't know the contents of the Debian package.

**Reply by hack3rcon** [reply]
Hello,
Thanks again.
Have you ever installed sysstat on Debian?
You can see the contents of the package by visiting [https://packages.debian.org/bookworm/sysstat](url)

**Reply by sysstat** [reply]
No, never installed on Debian.
Checking the link above, I can see that Debian actually uses systemd and so, starting `sysstat` and `sysstat-collect` services should be enough to collect system activities.

**Reply by hack3rcon** [reply]
Hi,
Thanks again.
1- The **sysstat-collect** is inactive and cannot start. What is the function of this service?

2- How do I clear the logs once a month?

**Reply by sysstat** [reply]
The `sysstat-collect` unit is the main service that calls the data collector periodically.
The `HISTORY` variable from `/etc/sysconfig/sysstat` file will tell sysstat how many days worth of data to keep (see `sysstat(5)` manual page)

Remember that you can tell sysstat to install et set up all the boring things for you. If your Debian package doesn't work as expected, don't hesitate to download sysstat tarball, configure, compile and install it (uninstall your Debian package before). Everything will work out of the box, e.g. to tell sysstat to collect data every 5 minutes and keep one month worth of data logs, enter:
```
./configure history=28  --enable-install-cron cron_interval=5
make
sudo make install
```
You can also use `iconfig`, the interactive configuration script, to tune configuration and check all the possible options instead of the default `configure` command.

**Reply by hack3rcon** [reply]
Hello,
Thanks again.
The `sysstat-collect` is inactive:
```
# systemctl status sysstat-collect
○ sysstat-collect.service - system activity accounting tool
     Loaded: loaded (/lib/systemd/system/sysstat-collect.service; static)
     Active: inactive (dead) since Fri 2024-03-15 04:10:04 EDT; 3min 10s ago
TriggeredBy: ● sysstat-collect.timer
       Docs: man:sa1(8)
    Process: 119337 ExecStart=/usr/lib/sysstat/sa1 1 1 (code=exited, status=0/SUCCESS)
   Main PID: 119337 (code=exited, status=0/SUCCESS)
        CPU: 8ms

Mar 15 04:10:04 gitlab systemd[1]: Starting sysstat-collect.service - system activity accounting tool...
Mar 15 04:10:04 gitlab systemd[1]: sysstat-collect.service: Deactivated successfully.
Mar 15 04:10:04 gitlab systemd[1]: Finished sysstat-collect.service - system activity accounting tool.
```
And the `HISTORY` variable is in `/etc/sysstat/sysstat` file.
